### PR TITLE
Update the mmap error code

### LINKED ISSUE #643: Inaccurate mmap() return error code
[Linux manual ](https://linux.die.net/man/2/mmap#:~:text=mmap%20%28%29%20creates%20a%20new%20mapping%20in%20the,most%20portable%20method%20of%20creating%20a%20new%20mapping.) describes the error code mmap syscall might return. The current Mystikos implementation returns -1 in many cases. The [code](https://github.com/deislabs/mystikos/blob/d0dcb081972ab5ac0ef34acdedaf36f4c3ee749f/kernel/mmanutils.c#L197) has an ATTN comment of "fix the return type". The return value coming form lower layer code also seems to be inaccurate in certain situation.

### PR DESCRIPTION
**Pull request by mingweishih**
Fixes #643

Signed-off-by: Ming-Wei Shih <mishih@microsoft.com>

### CODE REVIEW

File: kernel/mmanutils.c [hunk 1]
```diff
@@ -194,106 +194,127 @@ static ssize_t _map_file_onto_memory(
...
     }
 
     if (fd >= 0 && addr)
     {
         ssize_t n;
+
         // ATTN: call mmap or mremap here so that this range refers to
         // a mapped region.
-        // ATTN: is non-page-aligned length valid?
+        // ATTN: Use EACCES for failures related to fd access. This may not
```
Comment (bodzhang) [inline kernel/mmanutils.c:262]: LTP mmap15.c test tries to mmap to an address typically outside of app's linear address space -  `SYS_mmap(addr=fffffffffffff000 length=4096(1000) prot=1 flags=17 fd=3 offset=0)`, and expects EINVAL or ENOMEM error. The implementation here probably should check the whether the request address range is out of bound, before attempting to read file data (with mprotect()  call to make sure the memory range is writable).

File: kernel/mmanutils.c [hunk 2]
```diff
@@ -194,106 +194,127 @@ static ssize_t _map_file_onto_memory(
...
+            // conform the Linux kernel behavior.
+            if (!(prot & MYST_PROT_WRITE))
+            {
+                if (myst_mman_mprotect(
+                        &_mman, (void*)ret, length, prot | MYST_PROT_WRITE))
+                    ERAISE(-EACCES);
+            }
+            if ((n = _map_file_onto_memory(
+                     fd, offset, (void*)ret, length, flags)) < 0)
+                ERAISE((unsigned long)-n);
```
Comment (bodzhang) [inline kernel/mmanutils.c:302]: `n` is negative, shouldn't it be `ERAISE(n)`?

File: kernel/mmanutils.c (hunk 1 above)
Comment (mingweishih) [reply kernel/mmanutils.c]: `myst_mman_mprotect` have done such checks and return EINVAL when the checks failed. I updated the PR to cover the case when `myst_mman_mprotect` is not called.

File: kernel/mmanutils.c (hunk 2 above)
Comment (mingweishih) [reply kernel/mmanutils.c]: updated.

**Review by bodzhang — APPROVED** [review]

**Pull request merged**